tokio/src/runtime/task/list.rs

```diff
@@ -73,13 +58,23 @@ struct OwnedTasksInner<S: 'static> {
 }
 
 impl<S: 'static> OwnedTasks<S> {
-    pub(crate) fn new() -> Self {
+    /// grain must be an integer power of 2
+    pub(crate) fn new(grain: u32) -> Self {
+        assert_eq!(
+            grain & (grain - 1),
+            0,
+            "the grain of OwnedTasks must be an integer power of 2"
+        );
```

Comment:
it looks like `OwnedTasks::new` is only ever called with the values 1 and 16. What do you think about making this constructor private, and having the `pub(crate)` constructors be `new_current_thread` and `new_multi_thread` or something? that way, the value (which feels like an internal implementation detail) isn't leaked to the places where the `OwnedTasks` is constructed, and we can change the value in one place, rather than having to update _every_ call to `OwnedTasks::new`?

if we did that, we could probably also remove the assertion, or change it to a `debug_assert!`, since the caller can't pass in a value that isn't a power of two.
```suggestion
    pub(crate) fn new_current_thread() -> Self {
        Self::new(1)
    }
    
    pub(crate) fn new_multi_thread() -> Self {
        Self::new(16)
    }

    /// grain must be an integer power of 2
    fn new(grain: u32) -> Self {
        debug_assert_eq!(
            grain & (grain - 1),
            0,
            "the grain of OwnedTasks must be an integer power of 2"
        );
```

Reply:
I would make it configurable in `Builder` of the Runtime.